Dockerfile

```diff
@@ -1,60 +1,62 @@
 ENV DEBIAN_FRONTEND=noninteractive
 ENV WEBUI_VERSION=01
-ENV BASE_DIR="/opt/stable-diffusion"
-ENV SD_INSTALL_DIR="/opt/sd-install"
-ENV SD01_DIR="${BASE_DIR}/01-easy-diffusion"
-ENV SD02_DIR="${BASE_DIR}/02-sd-webui"
-ENV SD03_DIR="${BASE_DIR}/03-invokeai"
-ENV SD04_DIR="${BASE_DIR}/04-SD-Next"
-ENV SD05_DIR="${BASE_DIR}/05-comfy-ui"
-ENV SD06_DIR="${BASE_DIR}/06-Fooocus"
-ENV SD07_DIR="${BASE_DIR}/07-StableSwarm"
-ENV SD08_DIR="${BASE_DIR}/08-voltaML"
-ENV SD20_DIR="${BASE_DIR}/20-kubin"
-ENV SD50_DIR="${BASE_DIR}/50-lama-cleaner"
-ENV SD51_DIR="${BASE_DIR}/51-facefusion"
-ENV SD70_DIR="${BASE_DIR}/70-kohya"
-ENV XDG_CACHE_HOME="${BASE_DIR}/temp"
-
-# Installing required packages
+ENV BASE_DIR=/config
+ENV SD_INSTALL_DIR=/opt/sd-install
+ENV SD01_DIR=/config/01-easy-diffusion
+ENV SD02_DIR=/config/02-sd-webui
+ENV SD03_DIR=/config/03-invokeai
+ENV SD04_DIR=/config/04-SD-Next
+ENV SD05_DIR=/config/05-comfy-ui
+ENV SD06_DIR=/config/06-Fooocus
+ENV SD07_DIR=/config/07-StableSwarm
+ENV SD08_DIR=/config/08-voltaML
+ENV SD20_DIR=/config/20-kubin
+ENV SD50_DIR=/config/50-lama-cleaner
+ENV SD51_DIR=/config/51-facefusion
+ENV SD70_DIR=/config/70-kohya
+ENV XDG_CACHE_HOME=/config/temp
+
 RUN apt-get update -y -q=2 && \
-    apt-get install -y -q=2 curl wget mc nano rsync libgl1-mesa-glx libtcmalloc-minimal4 libcufft10 cmake build-essential python3-opencv libopencv-dev dotnet-sdk-7.0
-#    apt-get install -y -q=2 curl unzip bzip2 gpg mc nano rsync wget python3 python3-tk python3-venv git python3-pip build-essential ffmpeg \
-#        libglew-dev libglfw3-dev libglm-dev libgl1-mesa-glx python3-opencv libopencv-dev \
-#        libao-dev libmpg123-dev google-perftools
-RUN apt-get clean && \
-    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
-	
-# Creating necessary directories
-RUN mkdir -p \
-    ${XDG_CACHE_HOME} \
-    ${SD_INSTALL_DIR} \
-    /outputs
-	
-# Adding necessary files
-ADD parameters/* ${SD_INSTALL_DIR}/parameters/
-ADD *.sh /
-
-# Making the entry script executable and changing ownership of directories
-RUN chmod +x /*.sh
-RUN chown -R 99:100 /opt
-RUN chown -R 99:100 /outputs
-RUN wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh && bash Miniconda3-latest-Linux-x86_64.sh -b -p /opt/miniconda3 && rm Miniconda3-latest-Linux-x86_64.sh
-
-# Specifying volumes
-VOLUME ${BASE_DIR}
-VOLUME /outputs
-VOLUME /home/diffusion/.cache
-
-# Exposing necessary port
-EXPOSE 9000
-
-# Running the entry script
-CMD ["/entry.sh"]
+    apt-get install -y -q=2 curl \
+    wget \
+    mc \
+    nano \
+    rsync \
+    libgl1-mesa-glx \
+    libtcmalloc-minimal4 \
+    libcufft10 \
+    cmake \
+    build-essential \
+    python3-opencv \
+    libopencv-dev \
+    dotnet-sdk-7.0 \
+    git
+
+RUN apt-get clean &&     rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+RUN mkdir -p ${BASE_DIR}\temp ${SD_INSTALL_DIR} /config/outputs
+
+ADD parameters/* /opt/sd-install/parameters/
+
+COPY --chown=abc:abc *.sh ./
```

Comment:
What is abc? I see it used a lot.

Reply:
`abc` is the non-root user set up in the [lsio base image.](https://github.com/linuxserver/docker-baseimage-alpine/blob/0e7a9b6fbdf7de58c6d7e3e02b2e84ae8201a0fd/Dockerfile#L86)

All lsio containers uses [s6](https://skarnet.org/software/s6-linux-init/) as the entrypoint which kicks off base scripts, followed by custom user scripts, finally ending with the script that starts whatever the app the container is built for. The lifecycle for a lsio docker image -> container -> run generally looks like this:

* Dockerfile layers all runs as `root`
  * Entrypoint -> s6 init
* Container starts
  * The UID/GID for `abc` are [dynamically set](https://github.com/linuxserver/docker-baseimage-alpine/blob/0e7a9b6fbdf7de58c6d7e3e02b2e84ae8201a0fd/root/etc/s6-overlay/s6-rc.d/init-adduser/run) based on ENVs passed into the container (`PUID` `PGID`) and persistent user directory `/config` is chown'd to abc.
  * Any [supporting service/config scripts](https://github.com/FoxxMD/stable-diffusion/blob/ea2600415ce129838fa729a9897de52e68632d5f/docker/root/etc/s6-overlay/s6-rc.d/init-chown/run) (included by docker image author) are executed as `root`
  * Any custom user scripts are executed as `root`
  * The main app is [run as `abc`](https://github.com/FoxxMD/stable-diffusion/blob/ea2600415ce129838fa729a9897de52e68632d5f/docker/root/etc/s6-overlay/s6-rc.d/svc-app/run)

All of the changes in this branch pertaining to chown'ing and `abc` are because in holaf's version `root` executes `entry.sh` but in lsio version `abc` runs `entry.sh`